CondTools/SiPhase2Tracker/plugins/SiPhase2OuterTrackerLorentzAngleWriter.cc

```diff
@@ -0,0 +1,158 @@
+      m_tag(iConfig.getParameter<std::string>("tag")),
+      m_value(iConfig.getParameter<double>("value")) {}
+
+SiPhase2OuterTrackerLorentzAngleWriter::~SiPhase2OuterTrackerLorentzAngleWriter() {
+  delete lorentzAngle;
+  edm::LogInfo("SiPhase2OuterTrackerLorentzAngleWriter")
+      << "SiPhase2OuterTrackerLorentzAngleWriter::~SiPhase2OuterTrackerLorentzAngleWriter" << std::endl;
+}
+
+//
+// member functions
+//
+
+// ------------ method called for each event  ------------
+void SiPhase2OuterTrackerLorentzAngleWriter::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup) {
+  using Phase2TrackerGeomDetUnit = PixelGeomDetUnit;
+  using namespace edm;
+  edm::LogInfo("SiPhase2OuterTrackerLorentzAngleWriter")
+      << "SiPhase2OuterTrackerLorentzAngleWriter::analyze " << std::endl;
+
+  // Database services (write)
+  edm::Service<cond::service::PoolDBOutputService> mydbservice;
+  if (!mydbservice.isAvailable()) {
+    edm::LogWarning("SiPhase2OuterTrackerLorentzAngleWriter") << "Service is unavailable" << std::endl;
+    return;
+  }
+
+  std::string tag = mydbservice->tag(m_record);
+  unsigned int irun = iEvent.id().run();
+  edm::LogVerbatim("SiPhase2OuterTrackerLorentzAngleWriter") << "Writing on tag : " << tag << std::endl;
+  edm::LogVerbatim("SiPhase2OuterTrackerLorentzAngleWriter") << "Usinng as IOV run : " << irun << std::endl;
+
+  // map to be filled
+  std::unordered_map<unsigned int, float> detsLAtoDB;
+
+  // Retrieve tracker topology from geometry
+  edm::ESHandle<TrackerTopology> tTopoHandle;
+  iSetup.get<TrackerTopologyRcd>().get(tTopoHandle);
+  const TrackerTopology* const tTopo = tTopoHandle.product();
+
+  // Retrieve old style tracker geometry from geometry
+  edm::ESHandle<TrackerGeometry> pDD;
+  iSetup.get<TrackerDigiGeometryRecord>().get(pDD);
+  edm::LogInfo("SiPhase2OuterTrackerLorentzAngleWriter")
+      << " There are " << pDD->detUnits().size() << " modules in this geometry." << std::endl;
+
+  for (auto const& det_u : pDD->detUnits()) {
+    const DetId detid = det_u->geographicalId();
+    uint32_t rawId = detid.rawId();
+    int subid = detid.subdetId();
+    if (detid.det() == DetId::Detector::Tracker) {
+      const Phase2TrackerGeomDetUnit* pixdet = dynamic_cast<const Phase2TrackerGeomDetUnit*>(det_u);
+      assert(pixdet);
+      LogDebug("SiPhase2OuterTrackerLorentzAngleWriter") << rawId << " is a " << subid << " det" << std::endl;
+      if (subid == StripSubdetector::TOB || subid == StripSubdetector::TID) {
+        LogDebug("SiPhase2OuterTrackerLorentzAngleWriter")
+            << "subdetector ID:" << subid << " layer:" << tTopo->layer(detid) << std::endl;
+        detsLAtoDB[rawId] = m_value;
+      }
+    }
+  }
+
+  edm::LogInfo("SiPhase2OuterTrackerLorentzAngleWriter")
+      << " There are " << detsLAtoDB.size() << " OT Lorentz Angle values assigned" << std::endl;
+
+  // SiStripLorentzAngle object
+  lorentzAngle = new SiPhase2OuterTrackerLorentzAngle();
```

Comment:
This is a potential memory leak: `new ` is called in `analyze()`, but `delete ` is only called in the destructor
there doesn't appear to be any reason for `lorentzAngle` to be a class member at all (it's also not correctly indicated as a class member with leading `m_`)
could it just be allocated statically? if not, prefer to use smart pointers